tools/serverpod_cli/lib/src/analyzer/entities/entity_parser/entity_parser.dart

```diff
@@ -200,7 +242,10 @@ class EntityParser {
     if (relationMap is! YamlMap) return null;
     parent = relationMap.nodes[Keyword.parent]?.value;
 
-    if (parent is! String) return null;
+    if (parent is String) return parent;
+
+    var type = documentContents.nodes[Keyword.type]?.value;
+    if (Checker.isIdType(type)) return null;
```

Comment:
Question - Seems like this changes the behavior of this method.

If parent is not a string and the type is not an id type then we might return parent, what ever type it is?

Reply:
Good catch! There should be a type check here to make sure it is a string.